### PR TITLE
Add search api support

### PR DESCRIPTION
**Pull request by amrita-shrestha**
This PR adds two methods related to search API
- Search resources in a global scale (support various patterns to search)
- Search resources within drive

### CODE REVIEW

File: src/Drive.php [hunk 1]
```diff
@@ -855,6 +855,42 @@ public function setPermissionExpiration(string $permissionId, ?\DateTimeImmutabl
...
+     * @throws InvalidResponseException
+     * @throws HttpException
+     * @throws \DOMException
+     * @throws SabreClientHttpException
+     * @throws BadRequestException
+     * @throws ConflictException
+     * @throws NotFoundException
+     * @throws InternalServerErrorException
+     */
+    public function searchResource(string $pattern, ?string $limit = null): array
```
Comment (individual-it) [inline src/Drive.php:875]: we should document what are valid values for `$pattern`

File: src/WebDavClient.php [hunk 2]
```diff
@@ -150,4 +151,53 @@ public function setCustomSetting(array $connectionConfig, string $accessToken):
...
+     * @param string $pattern
+     * @param string|null $limit
+     * @param string|null $url
+     * @return array<string, array<int, array<string, string|null|object>>>
+     * @throws HttpException
+     * @throws SabreClientException
+     * @throws SabreClientHttpException
+     * @throws \DOMException
+     */
+    public function sendReportRequest(string $pattern, ?string $limit, ?string $url = ''): array
```
Comment (individual-it) [inline src/WebDavClient.php:165]: Even if we need `$limit` as a string in the XML it feels weird to make this parameter in the function an string, I think it should be `int` and if the XML strictly needs a string, then cast it to string later.
Using `string` here will make it much more error-prone. Same applies to higher level functions

File: tests/integration/Owncloud/OcisPhpSdk/DriveTest.php [hunk 3]
```diff
@@ -696,4 +696,38 @@ public function testReceiverUpdatesDriveShareRoleNotEnoughPermission(): void
...
+     * @throws HttpException
+     * @throws NotFoundException
+     * @throws InternalServerErrorException
+     * @throws \Exception
+     */
+    public function testSearchResourceInsideDrive(): void
+    {
+        $this->drive->createFolder('myfolder');
+        $this->drive->uploadFile('myfolder/somefile.txt', 'some content');
+        $this->drive->uploadFile('somefile.txt', 'root content');
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/DriveTest.php:714]: also upload files that should not be found, to make sure the function not just returns all files

File: tests/integration/Owncloud/OcisPhpSdk/DriveTest.php [hunk 4]
```diff
@@ -696,4 +696,38 @@ public function testReceiverUpdatesDriveShareRoleNotEnoughPermission(): void
...
+            }
+            sleep(2);
+            $attempt++;
+        } while ($attempt < $maxAttempts);
+
+        $this->assertCount(
+            2,
+            $resources,
+            "Expected two resource but found " . count($resources),
+        );
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisPhpSdk/DriveTest.php:731]: also check if the correct files have been returned

File: src/Drive.php [hunk 5]
```diff
@@ -855,6 +855,42 @@ public function setPermissionExpiration(string $permissionId, ?\DateTimeImmutabl
...
+    {
+        $pattern .= " scope:" . $this->getId();
+        $webDavClient = $this->createWebDavClient();
+        $webDavClient->setCustomSetting($this->connectionConfig, $this->accessToken);
+
+        $responses = $webDavClient->sendReportRequest($pattern, $limit, '/remote.php/dav/spaces');
```
Comment (individual-it) [inline src/Drive.php:881]: why using `remote.php`? This is just historical route and should not be used in oCIS anymore

File: src/Ocis.php [hunk 6]
```diff
@@ -1124,4 +1126,41 @@ public function getSharedByMe(): array
...
+     * @throws InvalidResponseException
+     * @throws HttpException
+     * @throws \DOMException
+     * @throws SabreClientHttpException
+     * @throws BadRequestException
+     * @throws ConflictException
+     * @throws NotFoundException
+     * @throws InternalServerErrorException
+     */
+    public function searchResource(string $pattern, ?string $limit = null): array
```
Comment (individual-it) [inline src/Ocis.php:1147]: looks very similar to `Drive::searchResource`, can that be combined?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: src/Drive.php (hunk 5 above)
Comment (amrita-shrestha) [reply src/Drive.php:881]: need **remote.php** for api remote

**Review by individual-it — APPROVED** [review]

**Pull request merged**